### PR TITLE
Refactoring Quick Fixes

### PR DESCRIPTION
**Pull request by lucasborin**
It forces the Checks to be aware of the quick fix feature.

### CODE REVIEW

File: src/checks/y_check_constants_interface.clas.abap [hunk 1]
```diff
@@ -1,10 +1,11 @@
 CLASS y_check_constants_interface DEFINITION PUBLIC INHERITING FROM y_check_base CREATE PUBLIC .
   PUBLIC SECTION.
-    METHODS constructor .
+    METHODS constructor.
 
   PROTECTED SECTION.
     METHODS inspect_statements REDEFINITION.
     METHODS inspect_tokens REDEFINITION.
+    METHODS: add_check_quickfix REDEFINITION.
```
Comment (lucasborin) [inline src/checks/y_check_constants_interface.clas.abap:8]: unnecessary `:`

File: src/checks/y_check_cx_root_usage.clas.abap [hunk 2]
```diff
@@ -4,6 +4,7 @@ CLASS y_check_cx_root_usage DEFINITION PUBLIC INHERITING FROM y_check_base CREAT
 
   PROTECTED SECTION.
     METHODS inspect_tokens REDEFINITION.
+    METHODS: add_check_quickfix REDEFINITION.
```
Comment (lucasborin) [inline src/checks/y_check_cx_root_usage.clas.abap:7]: unnecessary `:`

File: src/checks/y_check_deprecated_classes.clas.abap [hunk 3]
```diff
@@ -4,6 +4,7 @@ CLASS y_check_deprecated_classes DEFINITION PUBLIC INHERITING FROM y_check_base
 
   PROTECTED SECTION.
     METHODS inspect_tokens REDEFINITION.
+    METHODS: add_check_quickfix REDEFINITION.
```
Comment (lucasborin) [inline src/checks/y_check_deprecated_classes.clas.abap:7]: unnecessary `:`

File: src/checks/y_check_omit_optional_exp.clas.abap [hunk 4]
```diff
@@ -23,35 +26,47 @@ CLASS y_check_omit_optional_exp IMPLEMENTATION.
...
     ENDLOOP.
   ENDMETHOD.
 
+
+  METHOD add_check_quickfix.
+    TRY.
+        new_quickfix( )->replace_by( p_new_code = ``
+                                     p_context = cl_ci_quickfix_abap_context=>create_from_scan_tokens( p_ci_scan = ref_scan
+                                                                                                       p_stmt_idx = statement_index
+                                                                                                       p_from_token = 2 ) ).
```
Comment (lucasborin) [inline src/checks/y_check_omit_optional_exp.clas.abap:66]: how about `method_a( method_b ( EXPORTING input = var ) ).`?

File: src/checks/y_check_prefer_is_not.clas.abap [hunk 5]
```diff
@@ -4,6 +4,7 @@ CLASS y_check_prefer_is_not DEFINITION PUBLIC INHERITING FROM y_check_base CREAT
 
   PROTECTED SECTION.
     METHODS inspect_tokens REDEFINITION.
+    METHODS: add_check_quickfix REDEFINITION.
```
Comment (lucasborin) [inline src/checks/y_check_prefer_is_not.clas.abap:7]: unnecessary `:`

File: src/checks/y_check_statement_coverage.clas.abap [hunk 6]
```diff
@@ -5,13 +5,14 @@ CLASS y_check_statement_coverage DEFINITION PUBLIC INHERITING FROM y_check_base
   PROTECTED SECTION.
     METHODS execute_check REDEFINITION.
     METHODS inspect_tokens REDEFINITION.
+    METHODS: add_check_quickfix REDEFINITION.
```
Comment (lucasborin) [inline src/checks/y_check_statement_coverage.clas.abap:8]: unnecessary `:`

File: src/checks/y_check_test_seam_usage.clas.abap [hunk 7]
```diff
@@ -4,13 +4,13 @@ CLASS y_check_test_seam_usage DEFINITION PUBLIC INHERITING FROM y_check_base CRE
 
   PROTECTED SECTION.
     METHODS inspect_tokens REDEFINITION.
+    METHODS: add_check_quickfix REDEFINITION.
```
Comment (lucasborin) [inline src/checks/y_check_test_seam_usage.clas.abap:7]: unnecessary `:`

File: src/checks/y_check_omit_optional_exp.clas.abap [hunk 8]
```diff
@@ -23,35 +26,47 @@ CLASS y_check_omit_optional_exp IMPLEMENTATION.
...
+      CATCH cx_ci_quickfix_failed.
+        RETURN.
+    ENDTRY.
+  ENDMETHOD.
```
Comment (lucasborin) [inline src/checks/y_check_omit_optional_exp.clas.abap:70]: There is a native quick fix in ADT already:

**Comment by larshp** [comment]
new abaplint rule, https://rules.abaplint.org/unnecessary_chaining/ you can try enabling it, tho there are usually bugs in the beginning :)

**Review by eugen-guenther-sap — APPROVED** [review]

**Pull request merged**